cloverage/test/cloverage/dependency_test.clj

```diff
@@ -66,34 +67,26 @@
                                 [parkour.util :refer [ignore-errors]])
                       (:import [java.io Closeable Writer]
                                [clojure.lang IObj]))
-                   :expected '[parkour.io.dseq
-                               #{clojure.core.protocols clojure.core.reducers parkour.conf
-                                 parkour.cstep parkour.wrapper parkour.mapreduce.source
-                                 parkour.io.dseq.mapred parkour.io.dseq.mapreduce
-                                 parkour.util}]}})
+                   :expected '#{clojure.core.protocols clojure.core.reducers parkour.conf
+                                parkour.cstep parkour.wrapper parkour.mapreduce.source
+                                parkour.io.dseq.mapred parkour.io.dseq.mapreduce
+                                parkour.util}}})
 
 (t/deftest test-dependency-extraction
-  (doall (for [[ns-name
-                {ns-form :ns-source expected :expected}] (seq ns-fixtures)]
-           (let [result (cd/dependency-libs ns-form)]
-            ;; wrap in seq to work around lack of LazySeq.toString
-             (t/is (= expected result)
-                   (str "Parsing " ns-name
-                        " should give " expected
-                        " but got " result))))))
+  (doseq [[ns-name {ns-form :ns-source, expected :expected}] ns-fixtures]
+    (t/testing ns-name
+      (with-redefs [source/ns-form (constantly ns-form)]
+        (t/is (= expected
+                 (#'cd/dependencies (symbol ns-name))))))))
 
 (t/deftest test-dependency-sort
-  (let [dep-lists [['first #{'fourth}]
-                   ['second #{'fourth 'first}]
-                   ['third #{}]
-                   ['fourth #{'fifth}]
-                   ['fifth #{}]]
-        result    (cd/dependency-sort dep-lists)
-        index-map (zipmap result (range))]
-    (t/is (not (nil? result)) "Dependency sort should not be nil.")
-    (doseq [[name deps] dep-lists]
-      (let [my-index (get index-map name)]
-        (doseq [dep deps]
-          (t/is (< (get index-map dep) my-index)
-                (str "Prerequisite " dep " of " name " should be loaded before,"
-                     " but isn't: " result)))))))
+  (t/is (= '[clojure.walk
+             clojure.template
+             clojure.string
+             clojure.stacktrace
+             clojure.test]
+           (cd/in-dependency-order '[clojure.stacktrace
+                                     clojure.string
+                                     clojure.template
+                                     clojure.test
+                                     clojure.walk]))))
```

Comment:
Since I was able to replace most of the logic in this namespace with stuff from `tools.namespace` I added tests for important stuff like `in-dependency-order` which we previously had no tests for.